app/models/episode.rb

```diff
@@ -19,11 +28,27 @@ def update(args)
 
   class << self
     def prepare_params(args)
-      raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
+      if args[:video] && args[:date]
+        raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
 
-      args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
-      args[:date] = Time.parse(args[:date]).utc
+        args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
+        args[:date] = Time.parse(args[:date]).utc
+      end
       args
     end
   end
+
+  private
+
+  def check_youtube_status(video)
+    video.live_broadcast_content == 'none' ? 'over' : video.live_broadcast_content
+  end
+
+  def video_exists?(video)
+    @video = Yt::Video.new id: video
+    @video.title
+  rescue Yt::Errors::NoItems => e
+    puts e.message
```

Comment:
puts? 🤣

Reply:
And why you rescue from this error?

Reply:
Puts was there for debugging purpose, removed :)

We need to rescue form this error, so as app doesn't crash when, for instance, video was deleted on youtube and app is trying to get its status. In this case YT gem doesn't return nil, it just throws an error which crashes the whole app. You may find Kirill comment above, where he asks me to handle this thing.